dialects/direct/passes/cps2ds.cpp

```diff
@@ -128,9 +129,9 @@ const Def* CPS2DS::rewrite_body_(const Def* def) {
 
     if (auto old_nom = def->isa_nom()) { return old_nom; }
     DefArray new_ops{def->ops(), [&](const Def* op) { return rewrite_body(op); }};
-    if (def->isa<Tuple>()) return world.tuple(new_ops, def->dbg());
+    if (def->isa<Tuple>()) return world.tuple(new_ops); // TODO just remove this line? there is rebuild below
```

Comment:
There were problems where the tuple would be rebuilt but the old type `def->type()` was (wrongly) assigned to the new tuple.
 If this issue/behaviour is fixed, the line can be removed after a test.

Reply:
This line is already gone in your `ho_codegen` branch, so we can leave it like this for now.